test/TwinGet.AutomationInterface.Test/TestUtils/TestProject.cs

```diff
@@ -0,0 +1,63 @@
+﻿// This file is licensed to you under MIT license.
+
+using Microsoft.Build.Construction;
+using static TwinGet.AutomationInterface.AutomationInterfaceConstants;
+
+namespace TwinGet.AutomationInterface.Test.TestUtils
+{
+    /// <summary>
+    /// A class that setup a test project in a temporary path.
+    /// Any other instances that access resources created from <see cref="TestProject"/> must be closed prior to its disposal, otherwise <see cref="TestProject"/> will not be able to fully clean up the temporary resources.
+    /// </summary>
+    internal class TestProject : IDisposable
+    {
+        private bool _disposedValue;
+        private readonly List<TestTwincatProject> _twincatProjects = [];
+        public string RootPath { get; private set; }
+        public string SolutionPath { get; }
+        public SolutionFile SolutionFile { get; }
+        public IReadOnlyList<TestTwincatProject> TwincatProjects { get => _twincatProjects; }
+
+        public TestProject()
+        {
+            RootPath = Path.Join(Path.GetTempPath(), Guid.NewGuid().ToString());
+            Directory.CreateDirectory(RootPath);
+
+            TwinGet.Utils.IO.Directory.CopyDirectory(
+    TestTwincatProjectConstants.s_testTwincatProject,
+    RootPath, FileShare.Read).Wait();
+
+            SolutionPath = Directory.GetFiles(RootPath, "*.sln", SearchOption.AllDirectories).First();
```

Comment:
I guess this should find the Solution everytime, or there should be a solution everytime?

Reply:
Adding some context, as I have seemed to typo and somehow created this unsensible sentence. I was talking about the .First(), as can we trust there is always a solution existing, so First() is safe to use?